### PR TITLE
fix: code improvements and cleanup

### PR DESCRIPTION
**Pull request by shanpriyan**
1. Added cleanup for `setTimeout` in useEffect.
2. Removed the useEffect which will be triggered based on the `removeId` state which will further call the `removeToast` and invoked the `removeToast` directly inside the setTimeout.
3. Code improvements
```diff
- if (!toastList) return null;
- if (toastList && !toastList.length) return null;
+ if (!toastList?.length) return null;
```
will return `null` for both falsy values(null,undefined) and empty array [ ]

### CODE REVIEW

**Review by harish-sethuraman — APPROVED** [review]
LGTM

**Review by bigint — APPROVED** [review]
LGTM 🚀

File: src/components/Toast/components/ToastPortal.tsx [hunk 1]
```diff
@@ -78,23 +77,17 @@ export const ToastPortal = forwardRef((props, ref) => {
...
+            const timerId = setTimeout(() => {
+                removeToast(id);
             }, autoCloseTime || 3000);
+            return () => clearTimeout(timerId);
         }
     }, [toastList]);
 
-    if (!toastList) return null;
-    if (toastList && !toastList.length) return null;
+    if (!toastList?.length) return null;
```
Comment (mittalchirag) [inline src/components/Toast/components/ToastPortal.tsx:90]: You can use `isEmpty` util.

File: src/components/Toast/components/ToastPortal.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const Toast = (props: ToastProps) => {
             background={colorConfig.background}
             autoCloseTime={autoCloseTime}
             fullWidth={fullWidth}
-            onClick={() => (dismissOnClick && removeToast && id ? removeToast(id) : null)}
+            onClick={() => (dismissOnClick && id ? removeToast?.(id) : null)}
```
[the remark this answers is not in the document]
Comment (shanpriyan) [reply src/components/Toast/components/ToastPortal.tsx:42]: Yes, thought the same

**Review by mittalchirag — APPROVED** [review]

**Pull request merged**